data_pipelines_cli/airbyte_utils.py

```diff
@@ -38,52 +45,92 @@ def find_config_file(env: str, config_name: str = "airbyte") -> pathlib.Path:
+            )
+
+        return workspaces[0].get("workspaceId")
+
     def create_update_connections(self) -> None:
         """Create and update Airbyte connections defined in config yaml file"""
-        if self.airbyte_config["connections"]:
-            [
-                self.create_update_connection(self.airbyte_config["connections"][connection])
-                for connection in self.airbyte_config["connections"]
-            ]
-            [task.update(self.env_replacer(task)) for task in self.airbyte_config["tasks"]]
-            self.update_file(self.airbyte_config)
-
-    def create_update_connection(self, connection_config: Dict[str, Any]) -> Any:
+        if not self.airbyte_config["connections"]:
+            return
+
+        workspace_id = self.airbyte_config.get("workspace_id")
+        if workspace_id is None:
+            echo_warning(
+                "workspace_id was not provided in the configuration file - "
+                "fetching the default one from Airbyte deployment"
+            )
+            workspace_id = self.get_default_workspace_id()
+
+        for connection in self.airbyte_config["connections"]:
+            self.create_update_connection(
+                connection_config=self.airbyte_config["connections"][connection],
+                workspace_id=workspace_id,
+            )
+
+        for task in self.airbyte_config["tasks"]:
+            task.update(self.env_replacer(task))
+
+        self.update_file(self.airbyte_config)
+
+    def create_update_connection(self, connection_config: Dict[str, Any], workspace_id: str) -> Any:
+        def configs_equal(
+            conf_a: Dict[str, Any], conf_b: Dict[str, Any], equality_fields: Iterable[str]
+        ) -> bool:
+            conn_a = {k: v for k, v in conf_a.items() if k in equality_fields}
+            conn_b = {k: v for k, v in conf_b.items() if k in equality_fields}
+            return conn_a == conn_b
+
         connection_config_copy = copy.deepcopy(connection_config)
+
         response_search = self.request_handler(
-            "connections/search",
-            {
-                "sourceId": connection_config_copy["sourceId"],
-                "destinationId": connection_config_copy["destinationId"],
-                "namespaceDefinition": connection_config_copy["namespaceDefinition"],
-                "namespaceFormat": connection_config_copy["namespaceFormat"],
-            },
+            "connections/list", data={"workspaceId": workspace_id}
         )
-        if not response_search["connections"]:
+
+        equality_fields = [
+            "sourceId",
+            "destinationId",
+            "namespaceDefinition",
+            "namespaceFormat",
+        ]
+
+        matching_connections = [
+            connection
+            for connection in response_search["connections"]
+            if configs_equal(connection_config_copy, connection, equality_fields)
+        ]
+
+        if not matching_connections:
```

Comment:
What if there will be 2 connections and one will be new? Will it work or try to update both?

Reply:
@p-pekala If we have 2 connections in our config (with different source/destination ids) and one of them is a new one, that should be fine - `create_update_connection` is invoked separately per each connection in the for loop.